### PR TITLE
Update client/server implementation in the autobahn tests

### LINKED ISSUE #5606: Autobahn test needs updating
The `_make_handler()` method was removed in this PR: https://github.com/aio-libs/aiohttp/pull/3939

But, is still used at: https://github.com/aio-libs/aiohttp/blob/master/tests/autobahn/server.py#L37

This needs updating, if anybody would like to take on this small task.

References to `aiohttp.ws_connect()` appear to also need updating:
https://github.com/aio-libs/aiohttp/blob/master/tests/autobahn/client.py#L9

### PR DESCRIPTION
**Pull request by wefatherley**
## What do these changes do?
Alters `autobahn` client and server tests by updating deprecated syntax

## Are there changes in behavior for the user?
No, there should not be

## Related issue number
Resolves #5606

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by wfatherley** [comment]
Used the `crosbario/autobahn-testsuite:0.8.2` Docker image.

File: tests/autobahn/client.py [hunk 1]
```diff
@@ -6,31 +6,32 @@
...
-                break
-
-    url = url + "/updateReports?agent=%s" % name
-    ws = await aiohttp.ws_connect(url)
-    await ws.close()
+    async with aiohttp.ClientSession() as session:
+        async with session.ws_connect(url + "/getCaseCount") as ws:
+            num_tests = int((await ws.receive()).data)
+            print("running %d cases" % num_tests)
+            await ws.close()
```
Comment (greshilov) [inline tests/autobahn/client.py]: `ws` object automatically closes after the exit from the `session.ws_connect` context manager. So maybe it makes sense to replace all of the `ws_connect / ws.close()` pairs with separate context managers, like so:

```python3
async with session.ws_connect(url + "/getCaseCount") as ws:
    num_tests = int((await ws.receive()).data)
    print("running %d cases" % num_tests)
    
for i in range(1, num_tests + 1):
    ...
    async with session.ws_connect(text_url) as ws:
        ...
 ```

File: tests/autobahn/client.py [hunk 2]
```diff
@@ -6,31 +6,32 @@
...
+
+            for i in range(1, num_tests + 1):
+                print("running test case:", i)
+                text_url = url + "/runCase?case=%d&agent=%s" % (i, name)
+                ws = await session.ws_connect(text_url)
+                while True:
```
Comment (greshilov) [inline tests/autobahn/client.py]: This `while` loop can be replaced with the  `__aiter__` call on the `ws` object. The `CLOSE` message will be handled by the library :) 

```python3
async for msg in ws:
    if msg.type == aiohttp.WSMsgType.TEXT:
        await ws.send_str(msg.data)
    elif msg.type == aiohttp.WSMsgType.BINARY:
        await ws.send_bytes(msg.data)
    else:
        break
```

**Comment by greshilov** [comment]
@webknjaz PTAL

File: tests/autobahn/client.py (hunk 1 above)
Comment (wfatherley) [reply tests/autobahn/client.py]: Not sure why I didn't notice the assignment to `ws` nested in the outer `ws` context, or why the testclient returned OK on all the test cases, but I've made these changes :)

File: tests/autobahn/client.py (hunk 2 above)
Comment (wfatherley) [reply tests/autobahn/client.py]: That is much cleaner, thank you

**Review by greshilov — APPROVED** [review]
Thanks!

**Review by webknjaz — APPROVED** [review]

**Comment by Dreamsorcerer** [comment]
What's happening with these tests? There was a build failure for Windows, so I tried rerunning them, but now they aren't running at all...

**Comment by webknjaz** [comment]
I'm curious if that's related with today's GHA outage

**Comment by webknjaz** [comment]
@wefatherley mind backporting this?

**Comment by wfatherley** [comment]
Sure thing

**Comment by wfatherley** [comment]
It seems swapping out the deprecated `_make_handler` syntax can be backported to 3.2, while swapping out deprecated `ws_connect` syntax can be backported to 3.0? Does the cherry picker handle this on it's own?

**Comment by webknjaz** [comment]
It only does `git cherry-pick -x` and that's about it. Normally, if automatic backporting fails, it means that there's manual adjustments to be made.

**Pull request merged**